pkgs/os-specific/linux/iwd/0001-agent-network_connect_psk-check-if-the-config-file-f.patch

```diff
@@ -0,0 +1,59 @@
+From eb2bee7720ca0611902f3910d3d47a48315d9929 Mon Sep 17 00:00:00 2001
+From: Maximilian Bosch <maximilian@mbosch.me>
+Date: Fri, 17 Jul 2020 21:52:17 +0200
+Subject: [PATCH] agent/network_connect_psk: check if the config-file for the
+ SSID points to a store path
+
+NixOS-specific patch. When declaratively configuring networks, `iwd`
+shouldn't modify their config files.
+---
+ src/dbus.c    |  6 ++++++
+ src/network.c | 14 +++++++++++++-
+ 2 files changed, 19 insertions(+), 1 deletion(-)
+
+diff --git a/src/dbus.c b/src/dbus.c
+index ceede5c..5bdd04b 100644
+--- a/src/dbus.c
++++ b/src/dbus.c
+@@ -114,6 +114,12 @@ struct l_dbus_message *dbus_error_not_supported(struct l_dbus_message *msg)
+ 					"Operation not supported");
+ }
+ 
++struct l_dbus_message *dbus_error_file_is_a_nix_store_Path(struct l_dbus_message *msg, char *ssid)
++{
++	return l_dbus_message_new_error(msg, IWD_SERVICE ".NotSupportNixOS",
++					"Cannot update SSID %s as it's configured in a file that points to a store path! Please update your NixOS config accordingly!", ssid);
++}
```

Comment:
This should probably be added to `dbus.h` to fix `-Wimplicit-function-declaration`.

Reply:
Fixed.